db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java

```diff
@@ -0,0 +1,95 @@
+package com.eventstore.dbclient;
+
+public abstract class AbstractPersistentSubscriptionSettings<T> {
+    private int checkpointAfterMs;
+    private boolean extraStatistics;
+    private boolean resolveLinks;
+    private boolean fromStart;
+    private boolean fromEnd;
+    private int historyBufferSize;
+    private int liveBufferSize;
+    private int maxCheckpointCount;
+    private int maxRetryCount;
+    private int maxSubscriberCount;
+    private int messageTimeoutMs;
+    private int minCheckpointCount;
+    private int readBatchSize;
+    private ConsumerStrategy strategy;
+
+    public AbstractPersistentSubscriptionSettings(int checkpointAfterMs, boolean extraStatistics, boolean resolveLinks,
+                                                  int historyBufferSize, int liveBufferSize, int maxCheckpointCount,
+                                                  int maxRetryCount, int maxSubscriberCount, int messageTimeoutMs,
+                                                  int minCheckpointCount, int readBatchSize, ConsumerStrategy strategy,
+                                                  boolean fromStart, boolean fromEnd) {
+        this.checkpointAfterMs = checkpointAfterMs;
+        this.extraStatistics = extraStatistics;
+        this.resolveLinks = resolveLinks;
+        this.historyBufferSize = historyBufferSize;
+        this.liveBufferSize = liveBufferSize;
+        this.maxCheckpointCount = maxCheckpointCount;
+        this.maxRetryCount = maxRetryCount;
+        this.maxSubscriberCount = maxSubscriberCount;
+        this.messageTimeoutMs = messageTimeoutMs;
+        this.minCheckpointCount = minCheckpointCount;
+        this.readBatchSize = readBatchSize;
+        this.fromStart = fromStart;
+        this.fromEnd = fromEnd;
+        this.strategy = strategy;
+    }
+
+    public int getCheckpointAfterMs() {
```

Comment:
**FIX:** Setting does not match the [RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R51). Plus, it lacks the proper default: 2 seconds.

Reply:
We could also have an overload that takes a duration without the `Ms` in the name

Reply:
I think that having both methods is confusing. The general convention is to not include `Ms` in the name as all our timeouts should be set in milliseconds.